tests/Tests/600__ai_apps/650__rhosak/650__rhosak.robot

```diff
@@ -49,12 +57,79 @@ Verify User Is Able to Create And Delete a Kafka Stream
 
+Verify User Is Able to Produce and Consume Events
+  [Tags]  Sanity  Tier2
+  ...     ODS-248  ODS-247  ODS-246  ODS-245  ODS-243  ODS-241  ODS-239
+  [Teardown]  Clean Up RHOSAK    stream_to_delete=${stream_name_test}
+  ...                            topic_to_delete=${topic_name_test}
+  ...                            sa_clientid_to_delete=${kafka_client_id}
+  Enable RHOSAK
+  Verify Service Is Enabled  ${rhosak_displayed_appname}
+  Launch OpenShift Streams for Apache Kafka From RHODS Dashboard Link
+  Login to HCC  ${SSO.USERNAME}  ${SSO.PASSWORD}
+  Maybe Skip RHOSAK Tour
+  Sleep  5
+  Wait Until Page Contains    Create Kafka instance
+  ## Create kafka stream
+  Create Kafka Stream Instance  stream_name=${stream_name_test}  stream_region=${stream_region_test}  cloud_provider=${cloud_provider_test}
+  Search Item By Name and Owner in RHOSAK Table  name_search_term=${stream_name_test}  owner_search_term=${SSO.USERNAME}
+  Wait Until Keyword Succeeds    450  1  Check Stream Status  target_status=Ready  target_stream=${stream_name_test}
+  ## Create service account
+  Click From Actions Menu  search_col=Name  search_value=${stream_name_test}  action=Connection
+  Wait Until Page Contains Element  xpath=//input[@aria-label="Bootstrap server"]
+  ${bootstrap_server}=  Get Element Attribute    xpath=//input[@aria-label="Bootstrap server"]  value
+  ${kafka_sa_creds}=  Create Service Account From Connection Menu  sa_description=${service_account_test}
+  ${kafka_client_id}=  Set Variable  ${kafka_sa_creds}[kafka_client_id]
+  ${kafka_client_secret}=  Set Variable  ${kafka_sa_creds}[kafka_client_secret]
+  ## Create topic
+  Enter Stream  stream_name=${stream_name_test}
+  Enter Stream Topics Section
+  # Wait For HCC Splash Page
+  Create Topic  topic_name_to_create=${topic_name_test}
+  Page Should Contain Element    xpath=//a[text()='${topic_name_test}']
+  ## Assign permissions to SA
+  Enter Stream Access Section
+  Assign Permissions To ServiceAccount in RHOSAK  sa_client_id=${kafka_client_id}  sa_to_assign=${service_account_test}
+  ...                                             topic_to_assign=${topic_name_test}  cg_to_assign=${consumer_group_test}
+
+  ## Spawn a notebook with env variables
+  Switch Window  title:Red Hat OpenShift Data Science Dashboard
+  Wait for RHODS Dashboard to Load
+  Launch JupyterHub Spawner From Dashboard
+  Wait Until Page Contains Element  xpath://input[@name="Standard Data Science"]
+  &{notebook_envs}=  Create Dictionary  KAFKA_BOOTSTRAP_SERVER=${bootstrap_server}  KAFKA_USERNAME=${kafka_client_id}
+  ...                                   KAFKA_PASSWORD=${kafka_client_secret}  KAFKA_TOPIC=${topic_name_test}
+  ...                                   KAFKA_CONSUMER_GROUP=${consumer_group_test}
+  Spawn Notebook With Arguments  image=s2i-generic-data-science-notebook  envs=&{notebook_envs}
+  Wait for JupyterLab Splash Screen  timeout=60
+  Maybe Select Kernel
+  ## clone JL notebooks from git and run
```

Comment:
This should be handled by `Spawn Notebook With Arguments` after #172 is merged

Reply:
> Looks good, I would just be careful about the overlap with #172

I can send a new PR to fix the overlap after both are merged. Do you think the overlap may create troubles during execution?

Reply:
Yes, if both the keyword and the TC are waiting for `Wait for JupyterLab Splash Screen` the TC will fail because the keyword will give back control after the splash has already appeared

Reply:
you're right. I would say, if this PR is merged before the #172, i will raise a PR for correction afterwords, otherwiise I will just commit it removing those 2 lines. What do you say?

Reply:
If it's ready to go right now just merge it and we can update it after #172

Reply:
yes it is ready. Just missing an approval

Reply:
PR job for #172 is running, so up to you. I can give the approval if you want to merge now.

Reply:
ok let's merge, I'll send a PR after that once 172 will be finished